pep-0710.rst

```diff
@@ -0,0 +1,359 @@
+multiple wheels might be considered by installers (possibly from different
+package indexes) and immediately finding out which wheel file was actually used
+during the installation might be helpful. This way, developers can use
+information about wheels to debug issues making sure the desired wheel was
+actually installed. Another use case could be tools reporting software
+installed, such as tools reporting a SBOM (Software Bill of Materials), that might
+give more accurate reports. Yet another use case could be reconstruction of the
+Python environment by pinning each installed package to a specific distribution
+artifact consumed from a Python package index.
+
+The motivation described in this PEP is an extension of those in :pep:`610`.  Besides
+stating information about packages installed using a direct URL, installers SHOULD
+record information also for packages installed from Python package indexes when
+identified by their name, and optionally their version.
+
+Specification
+=============
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHOULD”,
+“SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL”
+in this document are to be interpreted as described in :rfc:`2119`.
+
+The ``provenance_url.json`` file SHOULD be created in the ``*.dist-info``
+directory by installers when installing a :term:`Distribution Package`
+specified by name (and optionally by :term:`Version Specifier`).
+
+This file MUST NOT be created when installing a distribution package from a requirement
+specifying a direct URL reference (including a VCS URL).
+
+Only one of the files ``provenance_url.json`` and ``direct_url.json`` (from :pep:`610`),
+may be present in a given ``*.dist-info`` directory; installers MUST NOT add both.
+
+The ``provenance_url.json`` JSON file MUST be a dictionary, compliant with
+:rfc:`8259` and UTF-8 encoded.
+
+If present, it MUST contain exactly two keys. The first one is ``url``, with
+type ``string``.  The second key MUST be ``archive_info`` with a value defined
+below.
+
+The value of the ``url`` key MUST be the URL from which the distribution package was downloaded. If a wheel is
+built from a source distribution, the ``url`` value MUST be the URL from which
+the source distribution was downloaded. If a wheel is downloaded and installed directly,
+the ``url`` field MUST be the URL from which the wheel was downloaded.
+As in the :ref:`direct URL origin specification<packaging:direct-url>`, the ``url`` value
+MUST be stripped of any sensitive authentication information for security reasons.
+
+The user:password section of the URL MAY however be composed of environment
+variables, matching the following regular expression:
+
+.. code-block:: text
+
+    \$\{[A-Za-z0-9-_]+\}(:\$\{[A-Za-z0-9-_]+\})?
+
+Additionally, the user:password section of the URL MAY be a well-known,
+non-security sensitive string. A typical example is ``git`` in the case of an
+URL such as ``ssh://git@gitlab.com``.
```

Comment:
This probably does not belong here. It was originally taken from PEP-610 and Git URL will never be stored in the ``url`` as packages are installed using their names. Is there a use-case for stripping user:password in URLs here?

Reply:
Maybe private package indices using HTTP Basic Authentication? But I'm certainly not the expert here, of course... @dstufft ?

Reply:
As I mentioned before, this whole section should be replaced by a reference to https://packaging.python.org/en/latest/specifications/direct-url-data-structure/ and mentioning here only the parts that should differ (i.e. that only ArchiveInfo applies and possibly so additional fine tuning). 

This way, these questions would not even be raised in this context. Unless we want to evolve the specification of the direct URL data structure. Or unless we conclude the direct URL data structure is not applicable in this case, but I very much doubt it.

Reply:
Also, I know there have been comments about the valid hash algorithms etc, but it might be that the person who commented about that was not aware of https://packaging.python.org/en/latest/specifications/direct-url-data-structure/ since it was not a standalone document at the time. It is also important to know that the hashes specification in the direct URL data structure is derived from PEP 691, so I think it is preferable to avoid creating yet another slightly different recommendation about acceptable hash algorithms.

Reply:
Yea, I also would assume you'd never record user:password.